src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,98 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = (props) => {
+  const { onAdd } = props;
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const inputs = [title, description, imgUrl, imdbUrl, imdbId];
+
+  const callbacks = [
+    setTitle,
+    setDescription,
+    setImgUrl,
+    setImdbId,
+    setImdbUrl,
+  ];
+
+  const isFormFilled = inputs
+    .map(input => input.trim())
+    .every(Boolean);
+
+  const clearForm = () => {
+    callbacks.forEach(callback => callback(''));
+  };
+
```

Comment:
some things I would do to improve the code
```suggestion
  const requiredInputs = [title, imgUrl, imdbUrl, imdbId];

  const setters = [
    setTitle,
    setDescription,
    setImgUrl,
    setImdbId,
    setImdbUrl,
  ];

  const isFormFilled = requiredInputs.every(input => !!input.trim())

  const clearForm = () => {
    setters.forEach(setter => setter(''));
  };

```